lib/uksgx/include/uk/sgx_cpu.h

```diff
@@ -0,0 +1,283 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef _UK_SGX_CPU_H_
+#define _UK_SGX_CPU_H_
+
+#include <uk/arch/types.h>
+#include <errno.h>
+
+#define	EX_TYPE_NONE			 0
+#define	EX_TYPE_DEFAULT			 1
+#define	EX_TYPE_FAULT			 2
+#define	EX_TYPE_UACCESS			 3
+#define	EX_TYPE_COPY			 4
+#define	EX_TYPE_CLEAR_FS		 5
+#define	EX_TYPE_FPU_RESTORE		 6
+#define	EX_TYPE_WRMSR			 7
+#define	EX_TYPE_RDMSR			 8
+#define	EX_TYPE_BPF			 9
+
+#define	EX_TYPE_WRMSR_IN_MCE		10
+#define	EX_TYPE_RDMSR_IN_MCE		11
+
+#define	EX_TYPE_DEFAULT_MCE_SAFE	12
+#define	EX_TYPE_FAULT_MCE_SAFE		13
+
+#define	BIT(nr)			        (1UL << (nr))
+
+#define __stringify_1(x...)	#x
+#define __stringify(x...)	__stringify_1(x)
+
+# define _ASM_EXTABLE_TYPE(from, to, type)			\
+	" .pushsection \"__ex_table\",\"a\"\n"			\
+	" .balign 4\n"						\
+	" .long (" #from ") - .\n"				\
+	" .long (" #to ") - .\n"				\
+	" .long " __stringify(type) " \n"			\
+	" .popsection\n"
+
+#define _ASM_EXTABLE(from, to)					\
+	_ASM_EXTABLE_TYPE(from, to, EX_TYPE_DEFAULT)
+
+/* Referred to as IA32_FEATURE_CONTROL in Intel's SDM. */
+#define X86_MSR_IA32_FEAT_CTL		            0x0000003a
+#define X86_FEAT_CTL_LOCKED				        BIT(0)
+#define X86_FEAT_CTL_VMX_ENABLED_INSIDE_SMX		BIT(1)
+#define X86_FEAT_CTL_VMX_ENABLED_OUTSIDE_SMX	BIT(2)
+#define X86_FEAT_CTL_SGX_LC_ENABLED			    BIT(17)
+#define X86_FEAT_CTL_SGX_ENABLED			    BIT(18)
+#define X86_FEAT_CTL_LMCE_ENABLED			    BIT(20)
+
+#define X86_MSR_IA32_EFER			            0xC0000080 /* extended feature enables */
+#define X86_EFER_SYSCALL			    		0x1UL
+#define X86_MSR_IA32_STAR			            0xC0000081 /* legacy mode SYSCALL target */
+#define X86_MSR_IA32_LSTAR			            0xC0000082 /* long mode SYSCALL target */
+#define X86_MSR_IA32_FMASK			            0xC0000084 /* legacy mode SYSCALL mask */
+
+/* Intel SGX Launch Enclave Public Key Hash MSRs */
+#define MSR_IA32_SGXLEPUBKEYHASH0	0x0000008C
+#define MSR_IA32_SGXLEPUBKEYHASH1	0x0000008D
+#define MSR_IA32_SGXLEPUBKEYHASH2	0x0000008E
+#define MSR_IA32_SGXLEPUBKEYHASH3	0x0000008F
+
+/* __cpuid_count(unsinged int info[4], unsigned int leaf, unsigned int subleaf); */
+#define __cpuid_count(x, y, z)                                                       \
+	asm volatile("cpuid"                                                   \
+		     : "=a"(x[0]), "=b"(x[1]), "=c"(x[2]), "=d"(x[3])          \
+		     : "a"(y), "c"(z))
+
+/* __cpuid(unsinged int info[4], unsigned int leaf); */
+#define __cpuid(x, y)                                                    \
+	asm volatile("cpuid"                                                   \
+		     : "=a"(x[0]), "=b"(x[1]), "=c"(x[2]), "=d"(x[3])          \
+		     : "a"(y))
+
+#define Genu 0x756e6547
+#define ineI 0x49656e69
+#define ntel 0x6c65746e
+
+static inline void rdmsr(unsigned int msr, __u32 *lo, __u32 *hi)
```

Comment:
You can use the functions already defined in `plat/common/include/x86/cpu.h`

Same for `rdmsrl`, `wrmsr`, `wrmsrl`